# Answering Business Questions Using SQL

In [1]:
%%capture
%load_ext sql
%sql sqlite:///chinook.db

'Connected: None@chinook.db'

### checking the different tables and views in our database

In [2]:
%%sql
SELECT
    name,
    type
FROM sqlite_master
WHERE type IN ("table","view")

Done.


[('album', 'table'),
 ('artist', 'table'),
 ('customer', 'table'),
 ('employee', 'table'),
 ('genre', 'table'),
 ('invoice', 'table'),
 ('invoice_line', 'table'),
 ('media_type', 'table'),
 ('playlist', 'table'),
 ('playlist_track', 'table'),
 ('track', 'table')]

In [3]:
%%sql
SELECT *
    FROM customer;

Done.


[(1, 'Luís', 'Gonçalves', 'Embraer - Empresa Brasileira de Aeronáutica S.A.', 'Av. Brigadeiro Faria Lima, 2170', 'São José dos Campos', 'SP', 'Brazil', '12227-000', '+55 (12) 3923-5555', '+55 (12) 3923-5566', 'luisg@embraer.com.br', 3),
 (2, 'Leonie', 'Köhler', None, 'Theodor-Heuss-Straße 34', 'Stuttgart', None, 'Germany', '70174', '+49 0711 2842222', None, 'leonekohler@surfeu.de', 5),
 (3, 'François', 'Tremblay', None, '1498 rue Bélanger', 'Montréal', 'QC', 'Canada', 'H2G 1A7', '+1 (514) 721-4711', None, 'ftremblay@gmail.com', 3),
 (4, 'Bjørn', 'Hansen', None, 'Ullevålsveien 14', 'Oslo', None, 'Norway', '0171', '+47 22 44 22 22', None, 'bjorn.hansen@yahoo.no', 4),
 (5, 'František', 'Wichterlová', 'JetBrains s.r.o.', 'Klanova 9/506', 'Prague', None, 'Czech Republic', '14700', '+420 2 4172 5555', '+420 2 4172 5555', 'frantisekw@jetbrains.com', 4),
 (6, 'Helena', 'Holý', None, 'Rilská 3174/6', 'Prague', None, 'Czech Republic', '14300', '+420 2 4177 0449', None, 'hholy@gmail.com', 5),
 (7, 'Astrid', 'Gruber', None, 'Rotenturmstraße 4, 1010 Innere Stadt', 'Vienne', None, 'Austria', '1010', '+43 01 5134505', None, 'astrid.gruber@apple.at', 5),
 (8, 'Daan', 'Peeters', None, 'Grétrystraat 63', 'Brussels', None, 'Belgium', '1000', '+32 02 219 03 03', None, 'daan_peeters@apple.be', 4),
 (9, 'Kara', 'Nielsen', None, 'Sønder Boulevard 51', 'Copenhagen', None, 'Denmark', '1720', '+453 3331 9991', None, 'kara.nielsen@jubii.dk', 4),
 (10, 'Eduardo', 'Martins', 'Woodstock Discos', 'Rua Dr. Falcão Filho, 155', 'São Paulo', 'SP', 'Brazil', '01007-010', '+55 (11) 3033-5446', '+55 (11) 3033-4564', 'eduardo@woodstock.com.br', 4),
 (11, 'Alexandre', 'Rocha', 'Banco do Brasil S.A.', 'Av. Paulista, 2022', 'São Paulo', 'SP', 'Brazil', '01310-200', '+55 (11) 3055-3278', '+55 (11) 3055-8131', 'alero@uol.com.br', 5),
 (12, 'Roberto', 'Almeida', 'Riotur', 'Praça Pio X, 119', 'Rio de Janeiro', 'RJ', 'Brazil', '20040-020', '+55 (21) 2271-7000', '+55 (21) 2271-7070', 'roberto.almeida@riotur.gov.br', 3),
 (13, 'Fernanda', 'Ramos', None, 'Qe 7 Bloco G', 'Brasília', 'DF', 'Brazil', '71020-677', '+55 (61) 3363-5547', '+55 (61) 3363-7855', 'fernadaramos4@uol.com.br', 4),
 (14, 'Mark', 'Philips', 'Telus', '8210 111 ST NW', 'Edmonton', 'AB', 'Canada', 'T6G 2C7', '+1 (780) 434-4554', '+1 (780) 434-5565', 'mphilips12@shaw.ca', 5),
 (15, 'Jennifer', 'Peterson', 'Rogers Canada', '700 W Pender Street', 'Vancouver', 'BC', 'Canada', 'V6C 1G8', '+1 (604) 688-2255', '+1 (604) 688-8756', 'jenniferp@rogers.ca', 3),
 (16, 'Frank', 'Harris', 'Google Inc.', '1600 Amphitheatre Parkway', 'Mountain View', 'CA', 'USA', '94043-1351', '+1 (650) 253-0000', '+1 (650) 253-0000', 'fharris@google.com', 4),
 (17, 'Jack', 'Smith', 'Microsoft Corporation', '1 Microsoft Way', 'Redmond', 'WA', 'USA', '98052-8300', '+1 (425) 882-8080', '+1 (425) 882-8081', 'jacksmith@microsoft.com', 5),
 (18, 'Michelle', 'Brooks', None, '627 Broadway', 'New York', 'NY', 'USA', '10012-2612', '+1 (212) 221-3546', '+1 (212) 221-4679', 'michelleb@aol.com', 3),
 (19, 'Tim', 'Goyer', 'Apple Inc.', '1 Infinite Loop', 'Cupertino', 'CA', 'USA', '95014', '+1 (408) 996-1010', '+1 (408) 996-1011', 'tgoyer@apple.com', 3),
 (20, 'Dan', 'Miller', None, '541 Del Medio Avenue', 'Mountain View', 'CA', 'USA', '94040-111', '+1 (650) 644-3358', None, 'dmiller@comcast.com', 4),
 (21, 'Kathy', 'Chase', None, '801 W 4th Street', 'Reno', 'NV', 'USA', '89503', '+1 (775) 223-7665', None, 'kachase@hotmail.com', 5),
 (22, 'Heather', 'Leacock', None, '120 S Orange Ave', 'Orlando', 'FL', 'USA', '32801', '+1 (407) 999-7788', None, 'hleacock@gmail.com', 4),
 (23, 'John', 'Gordon', None, '69 Salem Street', 'Boston', 'MA', 'USA', '2113', '+1 (617) 522-1333', None, 'johngordon22@yahoo.com', 4),
 (24, 'Frank', 'Ralston', None, '162 E Superior Street', 'Chicago', 'IL', 'USA', '60611', '+1 (312) 332-3232', None, 'fralston@gmail.com', 3),
 (25, 'Victor', 'Stevens', None, '319 N. Frances Street', 'Madison', 'WI', 'USA', '53703', '+1 (608) 257-0597', N

In [4]:
%%sql

WITH 
    usa_tracks_sold AS
        (
            SELECT il.*
                FROM invoice_line il
            INNER JOIN invoice i ON il.invoice_id = i.invoice_id
            INNER JOIN customer c ON c.customer_id = i.customer_id
            WHERE c.country = "USA"
        )
        
SELECT
    g.name genre,
    COUNT(uts.invoice_line_id) tracks_sold,
    CAST(COUNT(uts.invoice_line_id) AS float) / 
    (
        SELECT
            COUNT(*) FROM usa_tracks_sold
    ) percentage_sold
    FROM usa_tracks_sold uts
    INNER JOIN track t ON uts.track_id = t.track_id
    INNER JOIN genre g ON g.genre_id = t.genre_id
    GROUP BY 1
    ORDER BY 2 DESC; 

Done.


[('Rock', 561, 0.5337773549000951),
 ('Alternative & Punk', 130, 0.12369172216936251),
 ('Metal', 124, 0.11798287345385347),
 ('R&B/Soul', 53, 0.05042816365366318),
 ('Blues', 36, 0.03425309229305423),
 ('Alternative', 35, 0.03330161750713606),
 ('Latin', 22, 0.02093244529019981),
 ('Pop', 22, 0.02093244529019981),
 ('Hip Hop/Rap', 20, 0.019029495718363463),
 ('Jazz', 14, 0.013320647002854425),
 ('Easy Listening', 13, 0.012369172216936251),
 ('Reggae', 6, 0.005708848715509039),
 ('Electronica/Dance', 5, 0.004757373929590866),
 ('Classical', 4, 0.003805899143672693),
 ('Heavy Metal', 3, 0.0028544243577545195),
 ('Soundtrack', 2, 0.0019029495718363464),
 ('TV Shows', 1, 0.0009514747859181732)]

based on the table above, it is advised to purchase:

1. Red Tone - Punk
2. Slim JIm Bites - Blues
3. Meteor and the Girls - Pop.

However, it should be noted that these three genres together make up less than **18 percent** of total sales. 

We can be on the look out for artists in the Rock genre which make up more than **53 percent** of the total sales

# Analyzing Employee Sales Performance

In [5]:
%%sql     

SELECT * 
    FROM employee
    LIMIT 10;

Done.


[(1, 'Adams', 'Andrew', 'General Manager', None, '1962-02-18 00:00:00', '2016-08-14 00:00:00', '11120 Jasper Ave NW', 'Edmonton', 'AB', 'Canada', 'T5K 2N1', '+1 (780) 428-9482', '+1 (780) 428-3457', 'andrew@chinookcorp.com'),
 (2, 'Edwards', 'Nancy', 'Sales Manager', 1, '1958-12-08 00:00:00', '2016-05-01 00:00:00', '825 8 Ave SW', 'Calgary', 'AB', 'Canada', 'T2P 2T3', '+1 (403) 262-3443', '+1 (403) 262-3322', 'nancy@chinookcorp.com'),
 (3, 'Peacock', 'Jane', 'Sales Support Agent', 2, '1973-08-29 00:00:00', '2017-04-01 00:00:00', '1111 6 Ave SW', 'Calgary', 'AB', 'Canada', 'T2P 5M5', '+1 (403) 262-3443', '+1 (403) 262-6712', 'jane@chinookcorp.com'),
 (4, 'Park', 'Margaret', 'Sales Support Agent', 2, '1947-09-19 00:00:00', '2017-05-03 00:00:00', '683 10 Street SW', 'Calgary', 'AB', 'Canada', 'T2P 5G3', '+1 (403) 263-4423', '+1 (403) 263-4289', 'margaret@chinookcorp.com'),
 (5, 'Johnson', 'Steve', 'Sales Support Agent', 2, '1965-03-03 00:00:00', '2017-10-17 00:00:00', '7727B 41 Ave', 'Calgary', 'AB', 'Canada', 'T3B 1Y7', '1 (780) 836-9987', '1 (780) 836-9543', 'steve@chinookcorp.com'),
 (6, 'Mitchell', 'Michael', 'IT Manager', 1, '1973-07-01 00:00:00', '2016-10-17 00:00:00', '5827 Bowness Road NW', 'Calgary', 'AB', 'Canada', 'T3B 0C5', '+1 (403) 246-9887', '+1 (403) 246-9899', 'michael@chinookcorp.com'),
 (7, 'King', 'Robert', 'IT Staff', 6, '1970-05-29 00:00:00', '2017-01-02 00:00:00', '590 Columbia Boulevard West', 'Lethbridge', 'AB', 'Canada', 'T1K 5N8', '+1 (403) 456-9986', '+1 (403) 456-8485', 'robert@chinookcorp.com'),
 (8, 'Callahan', 'Laura', 'IT Staff', 6, '1968-01-09 00:00:00', '2017-03-04 00:00:00', '923 7 ST NW', 'Lethbridge', 'AB', 'Canada', 'T1H 1Y8', '+1 (403) 467-3351', '+1 (403) 467-8772', 'laura@chinookcorp.com')]

In [6]:
%%sql

WITH                     
    customer_purchase AS
    (
        SELECT 
            c.support_rep_id,
            SUM(i.total) total
        FROM invoice i
        INNER JOIN customer c ON i.customer_id = c.customer_id
        GROUP BY i.customer_id
    )
    
SELECT 
    e.first_name || " " || e.last_name employee_name,
    e.birthdate,
    e.hire_date,
    SUM(cp.total) total_sales
    
FROM employee e
LEFT JOIN customer_purchase cp ON e.employee_id = cp.support_rep_id
WHERE e.title ='Sales Support Agent'
GROUP BY 1
    ORDER BY 4 DESC;

Done.


[('Jane Peacock', '1973-08-29 00:00:00', '2017-04-01 00:00:00', 1731.51),
 ('Margaret Park', '1947-09-19 00:00:00', '2017-05-03 00:00:00', 1584.0),
 ('Steve Johnson', '1965-03-03 00:00:00', '2017-10-17 00:00:00', 1393.92)]

The difference in sales target between the highest sales support Agent - Jane and lowest sales support agent Steve Johnson is less than **400 dollars** corresponding to **20 percent** of sales reflects the difference in hiring dates.

# Analyzing Sales by Country

In [7]:
%%sql

WITH 
    country_or_other AS
        (
        SELECT
            CASE
                WHEN (
                       SELECT COUNT(*)
                        FROM customer  
                        WHERE country = c.country
                    ) = 1   THEN "other"
                    ELSE c.country
                END AS country,
            
            c.customer_id,
            il.*
        
        FROM invoice_line il
        INNER JOIN invoice i ON i.invoice_id = il.invoice_id
        INNER JOIN customer c ON i.customer_id = c.customer_id
        ),
        
country_data_sales AS 
        (
            SELECT
                country,
                COUNT( DISTINCT customer_id) total_customers,
                SUM(unit_price) total_sales,
                SUM(unit_price) / COUNT(DISTINCT customer_id) 
                avg_sales_customer,
                SUM(unit_price) / COUNT(DISTINCT invoice_id) avg_order,
                CASE
                    WHEN country = "other" THEN 1
                    ELSE 0
                END AS sort
            
            FROM country_or_other 
            GROUP BY country
            ORDER BY sort ASC, total_sales DESC
        )
    
SELECT
    country,
    total_customers customers,
    total_sales,
    avg_sales_customer customer_lifetime_value,
    avg_order
    
    FROM country_data_sales;
    
    

Done.


[('USA', 13, 1040.490000000008, 80.03769230769292, 7.942671755725252),
 ('Canada', 8, 535.5900000000034, 66.94875000000043, 7.047236842105309),
 ('Brazil', 5, 427.68000000000245, 85.53600000000048, 7.011147540983647),
 ('France', 5, 389.0700000000021, 77.81400000000042, 7.781400000000042),
 ('Germany', 4, 334.6200000000016, 83.6550000000004, 8.161463414634186),
 ('Czech Republic', 2, 273.24000000000103, 136.62000000000052, 9.108000000000034),
 ('United Kingdom', 3, 245.52000000000078, 81.84000000000026, 8.768571428571457),
 ('Portugal', 2, 185.13000000000022, 92.56500000000011, 6.3837931034482835),
 ('India', 2, 183.1500000000002, 91.5750000000001, 8.72142857142858),
 ('other', 15, 1094.9400000000085, 72.99600000000056, 7.448571428571486)]

## Albums vs Individual Tracks

The Chinook store is setup in a way that allows custome rto make purchases in one of the two ways:
* purchase a whole album
* purchase a collection of one or more individual tracks

The store does not let customers purchase a whole album, and then add individual tracks to that same purchase ( unles they do that by choosing each track manually). When customers purchase albums they are charged the same price as if they purchases eachh of those tracks separately.

Management are currently considering changing their purchasing stratehy to save money . The strategy they are considering is to purchase only the most popular tracks from each album from record companies, instead of purchasing every track from an album.

Our task is to find out what percentage of purchases are individual tracks vs whole albums, so that management can use this data to understand the effect this decision might have on overall revenue.

In order to answer the question, we're going to have to identify whether each invoice has all the tracks from an album. WE can do this by getting the list of tracks from an invoice and comparing it to the list of tracks from an album.  
We can find the album to compare the purchase to by looking up the album that one of the purchased tracks belongs to .

**Note**: It is very common when you are performing an analysit to have 'edge cases' which prevent you from getting a  100% accurate answer to your question. In this instance, we have two edge cases to consider:
* Albums that have only one or two tracks are likely to be purchased by customers as part of a collection of individual tracks.
* Customers may decide to manually select every track from an album, and the add a few individual tracks from other albums to their purchase.  

In the first case, since our analysis is concerned with maximizing revenue we cans safely ignore albums consisting of only a few tracks. The company has previously done analyis to confirm that the second case does not happen often, so we can ignore this case also.

In [8]:
%%sql
WITH 

invoice_first_track AS
    (
    SELECT
        il.invoice_id invoice_id,
        MIN(il.track_id) first_track_id
        FROM invoice_line il
    GROUP BY 1
    )
    
SELECT
    album_purchase,
    COUNT(invoice_id) number_of_invoice,
    CAST(COUNT(invoice_id) AS float) / (SELECT
                                           COUNT(*) 
                                        FROM invoice) percent
    
    FROM
        (SELECT
            ifs.*,
            CASE
                 WHEN 
                     (SELECT
                         t.track_id 
                      FROM track t
                          WHERE t.album_id = (SELECT
                                        t2.album_id 
                                        FROM track t2
                                WHERE t2.track_id = ifs.first_track_id)
                     EXCEPT
                     
                    SELECT
                        il2.track_id track_id 
                    FROM invoice_line il2
                        WHERE il2.invoice_id = ifs.invoice_id) IS NULL
                 
                AND 
                     (SELECT
                          il2.track_id track_id 
                     FROM invoice_line il2
                         WHERE il2.invoice_id = ifs.invoice_id
                     
                     EXCEPT
                      
                     SELECT
                         t.track_id 
                     FROM track t
                         WHERE t.album_id = (SELECT
                                        t2.album_id 
                                        FROM track t2
                                WHERE t2.track_id = ifs.first_track_id)
                      ) IS NULL
                 THEN "yes"
                 ELSE "no"
                 END AS "album_purchase"
         
         FROM
        invoice_first_track ifs
        )
        GROUP BY album_purchase;
    
    
    

Done.


[('no', 500, 0.8143322475570033), ('yes', 114, 0.18566775244299674)]

Based on the table above, focusing on select tracks from albums may represent a cost effective strategy. However, this will lead to almost a fifth drop in revenue.

# Which Artist is used in the most Playlists.

In [9]:
%%sql

with track_artist_name as 
            (
                select t.track_id, ar.name artist
                    from track t
                    left join album al on al.album_id =
                        t.album_id
                    inner join artist ar on ar.artist_id
                        = al.artist_id
            ),
    
track_playlist as 
            ( 
                select pl.name playlist, pt.playlist_id,
                pt.track_id 
                    from playlist_track pt
                    left join playlist pl on pl.playlist_id
                    = pt.playlist_id
            ),
    
playlist_track_id as 
            (
    
            select tp.playlist, tp.playlist_id, tp.track_id,
            ta.artist
                from track_playlist tp
                left join track_artist_name ta on 
                    ta.track_id = tp.track_id
            )
    
    select  artist,count(playlist) no_of_playlist,
           sum(no_of_tracks) no_of_tracks
        
        from (
            
                select DISTINCT(playlist), artist , 
                count(track_id) no_of_tracks
                from playlist_track_id
                GROUP BY playlist_id, artist
            )
        GROUP BY artist    
        ORDER BY no_of_tracks DESC, no_of_playlist DESC;

Done.


[('Iron Maiden', 3, 303),
 ('U2', 2, 198),
 ('Metallica', 3, 184),
 ('Led Zeppelin', 2, 138),
 ('Deep Purple', 2, 134),
 ('Pearl Jam', 3, 110),
 ('Eric Clapton', 3, 97),
 ('Faith No More', 2, 93),
 ('Lost', 1, 92),
 ('Lenny Kravitz', 2, 86),
 ('Red Hot Chili Peppers', 2, 80),
 ('Titãs', 2, 76),
 ('Chico Science & Nação Zumbi', 2, 72),
 ("Guns N' Roses", 2, 72),
 ('Os Paralamas Do Sucesso', 2, 70),
 ('The Rolling Stones', 2, 70),
 ('Van Halen', 2, 70),
 ('Kiss', 2, 69),
 ('Nirvana', 3, 64),
 ('Jamiroquai', 2, 64),
 ('Legião Urbana', 2, 62),
 ('Djavan', 3, 57),
 ('Foo Fighters', 2, 57),
 ('Various Artists', 1, 56),
 ('R.E.M.', 2, 55),
 ('Chico Buarque', 3, 53),
 ('The Office', 1, 53),
 ('Smashing Pumpkins', 2, 51),
 ('Gilberto Gil', 3, 50),
 ('The Tea Party', 2, 50),
 ('Green Day', 2, 49),
 ('Queen', 2, 49),
 ('Santana', 2, 48),
 ('Ozzy Osbourne', 3, 47),
 ('Cidade Negra', 2, 45),
 ('Caetano Veloso', 3, 42),
 ('Audioslave', 2, 41),
 ('James Brown', 2, 40),
 ('Creedence Clearwater Revival', 1, 40),
 ('House Of Pain', 2, 38),
 ('Miles Davis', 2, 38),
 ('Zeca Pagodinho', 2, 38),
 ('Tim Maia', 2, 37),
 ('The Clash', 2, 36),
 ('Mötley Crüe', 3, 35),
 ('Body Count', 2, 34),
 ('Skank', 2, 34),
 ('The Cult', 2, 34),
 ('Antônio Carlos Jobim', 2, 33),
 ('Spyro Gyra', 2, 33),
 ('Soundgarden', 3, 32),
 ('Cássia Eller', 2, 32),
 ('Def Leppard', 2, 32),
 ('Funk Como Le Gusta', 2, 32),
 ('Planet Hemp', 2, 32),
 ('Gonzaguinha', 3, 30),
 ('Aerosmith', 2, 30),
 ('Milton Nascimento', 2, 29),
 ('Ed Motta', 2, 28),
 ('Passengers', 2, 28),
 ('Rush', 2, 28),
 ('The Police', 2, 28),
 ('UB40', 2, 28),
 ('Lulu Santos', 1, 28),
 ('Alanis Morissette', 2, 26),
 ('Incognito', 2, 26),
 ('Alice In Chains', 3, 25),
 ('Stone Temple Pilots', 3, 25),
 ('BackBeat', 2, 24),
 ('Jota Quest', 2, 24),
 ('Page & Plant', 2, 24),
 ('Battlestar Galactica (Classic)', 1, 24),
 ('Frank Sinatra', 1, 24),
 ('Amy Winehouse', 1, 23),
 ('Heroes', 1, 23),
 ('Bruce Dickinson', 2, 22),
 ('R.E.M. Feat. Kate Pearson', 2, 22),
 ('Gene Krupa', 1, 22),
 ('Temple of the Dog', 3, 21),
 ('Joe Satriani', 2, 20),
 ('Raimundos', 2, 20),
 ('Battlestar Galactica', 1, 20),
 ('The Who', 1, 20),
 ('AC/DC', 2, 19),
 ('Black Sabbath', 2, 19),
 ('Falamansa', 2, 19),
 ('The Black Crowes', 1, 19),
 ('Black Label Society', 1, 18),
 ('Marisa Monte', 1, 18),
 ('Marvin Gaye', 1, 18),
 ('Elis Regina', 2, 17),
 ('Jorge Ben', 2, 17),
 ('Judas Priest', 2, 17),
 ('Motörhead', 2, 17),
 ('Jimi Hendrix', 1, 17),
 ('Marcos Valle', 1, 17),
 ('O Rappa', 1, 17),
 ('Apocalyptica', 2, 16),
 ('O Terço', 1, 15),
 ('Toquinho & Vinícius', 1, 15),
 ('Vinícius De Moraes', 1, 15),
 ('Terry Bozzio, Tony Levin & Steve Stevens', 2, 14),
 ('Chris Cornell', 1, 14),
 ('João Suplicy', 1, 14),
 ('Olodum', 1, 14),
 ('Os Mutantes', 1, 14),
 ('Raul Seixas', 1, 14),
 ('Scorpions', 2, 13),
 ('JET', 1, 13),
 ('Velvet Revolver', 1, 13),
 ('Eugene Ormandy', 6, 12),
 ('David Coverdale', 1, 12),
 ('Godsmack', 1, 12),
 ('Mônica Marianno', 1, 12),
 ('Accept', 3, 11),
 ('Buddy Guy', 1, 11),
 ('System Of A Down', 1, 11),
 ('The Doors', 1, 11),
 ('Berliner Philharmoniker & Herbert Von Karajan', 5, 10),
 ('Cláudio Zoli', 1, 10),
 ('Marillion', 1, 10),
 ('Men At Work', 1, 10),
 ("Paul D'Ianno", 1, 10),
 ('Stevie Ray Vaughan & Double Trouble', 1, 10),
 ('Dennis Chambers', 1, 9),
 ('Frank Zappa & Captain Beefheart', 1, 9),
 ('Pink Floyd', 1, 9),
 ('Academy of St. Martin in the Fields & Sir Neville Marriner', 5, 8),
 ("The King's Singers", 5, 8),
 ('Michael Tilson Thomas & San Francisco Symphony', 4, 8),
 ('Billy Cobham', 1, 8),
 ('English Concert & Trevor Pinnock', 5, 7),
 ('Academy of St. Martin in the Fields, Sir Neville Marriner & Thurston Dart', 4, 4),
 ('Adrian Leaper & Doreen de Feis', 4, 4),
 ('Alberto Turco & Nova Schola Gregoriana', 4, 4),
 ('Antal Doráti & London Symphony Orchestra', 4, 4),
 ('Barry Wordsworth & BBC Concert Orchestra', 4, 4),
 ('Berliner Philharmoniker & Hans Rosbaud', 4, 4),
 ('Boston Symphony Orchestra & Seiji Ozawa', 4, 4),
 (

Artist <font color='red'> Iron maiden </font> has the most feature of 303 tracks across 3 playlists with 34% more feature than the next highest artist <font color='blue'>U2 </font>

Note: certain playlist Id had repeated thesame tracks and number of tracks like playlist_id 1,8,3,10. For these playlist Id with thesame tracks, only one duplicate was dropped.